lib/protobuf/field/base_field.rb

```diff
@@ -172,16 +172,19 @@ def wire_type
       #
 
       def define_accessor(simple_field_name, fully_qualified_field_name)
-        message_class.class_eval do
-          define_method("#{simple_field_name}!") do
-            @values[fully_qualified_field_name]
+        message_class.class_eval <<-ruby, __FILE__, __LINE__
+          def #{simple_field_name}!
+            @values[:#{fully_qualified_field_name}]
           end
-        end
 
-        message_class.class_eval do
-          define_method(simple_field_name) { self[fully_qualified_field_name] }
-          define_method("#{simple_field_name}=") { |v| self[fully_qualified_field_name] = v }
-        end
+          def #{simple_field_name}
+            self[:#{fully_qualified_field_name}]
+          end
+
+          def #{simple_field_name}=(value)
+            self[:#{fully_qualified_field_name}] = value
```

Comment:
Updated. Now that the full qualified field name code is in master, I rebased to see how the current implementation would work and got errors because `:some.thing` is not valid ruby. Change this to `:"some.thing"` works just fine.